tools/storyteller/StoryReader.java

```diff
@@ -21,13 +30,86 @@ public StoryReader(FileUtils fileUtils) {
     this.fileUtils = fileUtils;
   }
 
-  public ImmutableList<Story> getStories(String path) {
-    if (fileUtils.fileExists(path)) {
-      StoryList stories = (StoryList) fileUtils.readPrototxtUnchecked(path, StoryList.newBuilder());
-      return ImmutableList.copyOf(stories.getStoryList());
-    } else {
+  ImmutableList<Story> getUnsharedStories(String absPath, boolean loadScreenshots) {
+    return getStories(absPath, loadScreenshots, StoriesState.UNSHARED);
+  }
+
+  ImmutableList<Story> getSharedStories(String absPath, boolean loadScreenshots) {
+    return getStories(absPath, loadScreenshots, StoriesState.SHARED);
+  }
+
+  private ImmutableList<Story> getStories(
+      String absPath, boolean loadScreenshots, StoriesState state) {
+    List<String> absPaths = new ArrayList<>();
+    try {
+      absPaths = fileUtils
+          .listContents(absPath)
+          .stream()
+          .filter(
+              file -> file.endsWith(
+                      state.equals(
+                          StoriesState.UNSHARED)
+                          ? StorytellerConfig.STORIES_FILENAME
+                          : ".prototxt"))
+          .sorted()
+          .map(filename -> fileUtils.joinPaths(absPath, filename))
+          .collect(Collectors.toList());
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (absPaths.isEmpty()) {
       return ImmutableList.of();
+    } else {
+      StoryList.Builder storiesBuilder = StoryList.newBuilder();
```

Comment:
But we still need `StoryList` because it is used to read stories from prototxt file:
```
(StoryList) fileUtils.readPrototxtUnchecked(path, StoryList.newBuilder())
```

Reply:
Ok, you're right. I think it's actually better as it was before. Sorry.

Reply:
Ok, I actually think it was better the way you did it originally. Sorry.

Reply:
Reverted it.